src/Mvc/Mvc.Core/src/Infrastructure/JsonSerializerOptionsCopyConstructor.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Text.Encodings.Web;
+
+namespace System.Text.Json
+{
+    internal static class JsonSerializerOptionsCopyConstructor
+    {
+        public static JsonSerializerOptions Copy(this JsonSerializerOptions serializerOptions, JavaScriptEncoder encoder)
+        {
+            var copiedOptions = new JsonSerializerOptions
+            {
+                AllowTrailingCommas = serializerOptions.AllowTrailingCommas,
+                DefaultBufferSize = serializerOptions.DefaultBufferSize,
+                DictionaryKeyPolicy = serializerOptions.DictionaryKeyPolicy,
+                IgnoreNullValues = serializerOptions.IgnoreNullValues,
+                IgnoreReadOnlyProperties = serializerOptions.IgnoreReadOnlyProperties,
+                MaxDepth = serializerOptions.MaxDepth,
+                PropertyNameCaseInsensitive = serializerOptions.PropertyNameCaseInsensitive,
+                PropertyNamingPolicy = serializerOptions.PropertyNamingPolicy,
+                ReadCommentHandling = serializerOptions.ReadCommentHandling,
+                WriteIndented = serializerOptions.WriteIndented
+            };
+
+            for (var i = 0; i < serializerOptions.Converters.Count; i++)
```

Comment:
```suggestion
            for (var converterIdx = 0; converterIdx < serializerOptions.Converters.Count; converterIdx++)
```

Reply:
I'm not sure this makes the code any better.